core/trino-parser/src/main/java/io/trino/sql/tree/ArithmeticBinaryExpression.java

```diff
@@ -45,6 +45,7 @@ public String getValue()
     private final Operator operator;
     private final Expression left;
     private final Expression right;
+    private int hash;
```

Comment:
that probably should be volatile too

Reply:
no, it shouldn't be volatile.  volatile may force read from memory

see also `java.lang.String#hash` field (not volatile)